### PR TITLE
CPR-231 Add domain event listener for prisoners

### PR DESCRIPTION
**Pull request by H-Iwanejko**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/PrisonerDomainEventsListener.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.context.ApplicationContext
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.personrecord.message.processors.IEventProcessor
+import uk.gov.justice.digital.hmpps.personrecord.model.DomainEvent
+import uk.gov.justice.digital.hmpps.personrecord.model.SQSMessage
+
+const val PRISONER_EVENTS_QUEUE_CONFIG_KEY = "cprnomiseventsqueue"
+
+@Component
+class PrisonerDomainEventsListener(
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/PrisonerDomainEventsListener.kt:19]: Copied same as Offender, as will refactor in future PR's

File: src/main/resources/application-local.yml [hunk 2]
```diff
@@ -30,8 +30,14 @@ hmpps.sqs:
       dlqMaxReceiveCount: 1
     cprdeliusoffendereventsqueue:
       queueName: cpr_delius_offender_events_queue
+      dlqName: cpr_delius_offender_events_queue_dlq
       subscribeTopicId: domainevents
-      subscribeFilter: '{"eventType":[ "probation-case.engagement.created", "prisoner-offender-search.prisoner.created", "prisoner-offender-search.prisoner.updated"] }'
+      subscribeFilter: '{"eventType":[ "probation-case.engagement.created"] }'
```
Comment (markreesmoj) [inline src/main/resources/application-local.yml:35]: should have the other offender events we added this morning in this subscription (not related to this change I know)

File: src/test/resources/application-test.yml [hunk 3]
```diff
@@ -47,14 +47,17 @@ hmpps.sqs:
       queueName: cpr_delius_offender_events_queue
       dlqName: cpr_delius_offender_events_queue_dlq
       subscribeTopicId: domainevents
-      subscribeFilter: '{"eventType":[ "probation-case.engagement.created", "prisoner-offender-search.prisoner.created", "prisoner-offender-search.prisoner.updated"] }'
+      subscribeFilter: '{"eventType":[ "probation-case.engagement.created"] }'
```
Comment (markreesmoj) [inline src/test/resources/application-test.yml:50]: as per the previous comment - I will do this as part of my ticket

**Review by markreesmoj — APPROVED** [review]
Very nice

**Pull request merged**